kalite/distributed/tests/features/steps/exercises.py

```diff
@@ -0,0 +1,45 @@
+from behave import *
+from django.conf import settings
+from selenium.webdriver.common.by import By
+from selenium.common.exceptions import TimeoutException
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+TIMEOUT = 3
+
+@given("I open an exercise")
+def step_impl(context):
+    context.browser.get(context.browser_url("/"))
+
+@given("the exercise is available")
+def step_impl(context):
+    # NOTE(MCGallaspy): We really need a way to interact meaningfully with the learn page...
+    # It's hard for me and makes me sad. :(
+    #context.browser.get(context.browser_url("learn/some/random/exercise"))
+    pass
+
+@given("the exercise is not available")
+def step_impl(context):
+    # See above. This is hard for me to implement. :(
+    pass
+
+@then("I should see an alert")
```

Comment:
We should have a test expectation with:

```python
@then("I should fail non-deterministically")
def  step_impl(context):
    import random; assert random.choice([True, False])
```

Just so we can replicate our current test suite test-by-test.